ecommerce/extensions/payment/utils.py

```
@@ -66,6 +67,39 @@ def clean_field_value(value):
     return re.sub(r'[\^:"\']', '', value)
 
 
+def embargo_check(request, products):
+    """ Checks if the user has access to purchase products by calling the LMS embargo API.
+
+    Args:
+        request (object): The current request
+        courses (list): A list of courses to check access against
+
+    Returns:
+        True or False
+    """
+    courses = []
+    _, _, ip = parse_tracking_context(request.user)
+
+    for product in products:
+        courses.append(product.course.id)
+
+    params = {
+        'user': request.user,
+        'ip_address': ip,
+        'course_ids': courses
+    }
+
+    try:
+        response = request.site.siteconfiguration.embargo_api_client.course_access.get(**params)
+        if not response.get('access'):
+            return False
+    except:  # pylint: disable=bare-except
+        # We are going to allow purchase if the API is un-reachable.
```

I really want to catch all exceptions.